1.0rc1&2: test_eloc_attributes() fail on i386

This is another separation of #3380. `test_eloc_attributes` fails on i386:
```
    def test_eloc_attributes():
        from .. import AltAz, ITRS, GCRS, EarthLocation
    
        el = EarthLocation(lon=12.3*u.deg, lat=45.6*u.deg, height=1*u.km)
        it = ITRS(representation.SphericalRepresentation(lon=12.3*u.deg, lat=45.6*u.deg, distance=1*u.km))
        gc = GCRS(ra=12.3*u.deg, dec=45.6*u.deg, distance=6375*u.km)
    
        el1 = AltAz(location=el).location
        assert isinstance(el1, EarthLocation)
        assert el1.latitude == el.latitude
        assert el1.longitude == el.longitude
        assert el1.height == el.height
    
        el2 = AltAz(location=it).location
        assert isinstance(el2, EarthLocation)
        assert el2.latitude != it.spherical.lat
>       assert el2.longitude != it.spherical.lon
E       assert <Longitude [ 12.3] deg> != <Longitude 12.3 deg>
E        +  where <Longitude [ 12.3] deg> = <EarthLocation [(0.6836029704431764, 0.14904955169200265, 0.7144726796328034)] km>.longitude
E        +  and   <Longitude 12.3 deg> = <SphericalRepresentation lon=12.3 deg, lat=45.6 deg, distance=1.0 km>.lon
E        +    where <SphericalRepresentation lon=12.3 deg, lat=45.6 deg, distance=1.0 km> = <ITRS Coordinate (obstime=J2000.000): x=0.683602970443 km, y=0.149049551692 km, z=0.714472679633 km>.spherical
astropy/coordinates/tests/test_frames.py:531: AssertionError
```
(Update: I git confused with my logs. arm64 is not affected)

--- a/astropy/coordinates/tests/test_frames.py
+++ b/astropy/coordinates/tests/test_frames.py
@@ -552,20 +552,29 @@ def test_eloc_attributes():
 
     el1 = AltAz(location=el).location
     assert isinstance(el1, EarthLocation)
+    # these should match *exactly* because the EarthLocation
     assert el1.latitude == el.latitude
     assert el1.longitude == el.longitude
     assert el1.height == el.height
 
     el2 = AltAz(location=it).location
     assert isinstance(el2, EarthLocation)
-    assert el2.latitude != it.spherical.lat
-    assert el2.longitude != it.spherical.lon
+    # these should *not* match because giving something in Spherical ITRS is
+    # *not* the same as giving it as an EarthLocation: EarthLocation is on an
+    # elliptical geoid. So the longitude should match (because flattening is
+    # only along the z-axis), but latitude should not. Also, height is relative
+    # to the *surface* in EarthLocation, but the ITRS distance is relative to
+    # the center of the Earth
+    assert not np.allclose(el2.latitude, it.spherical.lat)
+    assert np.allclose(el2.longitude, it.spherical.lon)
     assert el2.height < -6000*u.km
 
-    el3 = AltAz(location=gc).location  #this one implicitly does transform_to
+    el3 = AltAz(location=gc).location
+    # GCRS inputs implicitly get transformed to ITRS and then onto
+    # EarthLocation's elliptical geoid. So both lat and lon shouldn't match
     assert isinstance(el3, EarthLocation)
-    assert el3.latitude != gc.dec
-    assert el3.longitude != gc.ra
+    assert not np.allclose(el3.latitude, gc.dec)
+    assert not np.allclose(el3.longitude, gc.ra)
     assert np.abs(el3.height) < 500*u.km
 
 

